### ISSUE TITLE
Regression in literal's encoding

### ISSUE THREAD

**Issue by lgrignon** [cincheo/jsweet#103]
Using utf-8 java file with String foo = "baré" ;
Outputs var foo: string = "bar\u00c3\u00a9" ;

Note: the bundle option is on, and e'coding option is set to UTF-8

**Comment by renaudpawlak** [comment]
I cannot reproduce that problem. More context?

**Comment by lgrignon** [comment]
I just reproduced it. 
Another context note is that I am using the `mvn clean generate-sources` command line.

**Comment by renaudpawlak** [comment]
Does it change something when adding the following property to your ``pom.xml``?

```xml
<project>
  ...
  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
  </properties>
  ...
</project>
```

**Comment by renaudpawlak** [comment]
Also, does it still do it if you tell JSweet to use UTF-8 encoding with the appropriate option?

```xml
<encoding>UTF-8</encoding>
```

**Comment by lgrignon** [comment]
nope, I already had it in my pom

**Comment by renaudpawlak** [comment]
The problem is that JSweet generates strings with unicode escape sequences in it rather than special characters such as accents. Using unicode is not bad per se (it is not really a bug), but what is bad is that we should preserve the original code intentions as much as possible, especially when the UTF-8 option is on. So this should be fixed at some point.

**Comment by lgrignon** [comment]
Problem isolated in SyntaxTest.literalsTest

`
$ workspace-jsweet\jsweet>mvn clean install -Dtest=SyntaxTests
`
[...]
```
ERROR: unclosed character literal at src\test\java\source\syntax\Literals.java(39)
ERROR: illegal character: '\u00a9' at src\test\java\source\syntax\Literals.java(39)
ERROR: unclosed character literal at src\test\java\source\syntax\Literals.java(39)
2016-05-20 12:46:07.007 INFO  JSweetTranspiler:612 - transpilation process finished in 52 ms
java.lang.Exception: unable to evaluate: transpilation errors remain
        at org.jsweet.transpiler.JSweetTranspiler.eval(JSweetTranspiler.java:483)
        at org.jsweet.transpiler.JSweetTranspiler.eval(JSweetTranspiler.java:369)
        at org.jsweet.test.transpiler.AbstractTest.eval(AbstractTest.java:198)
        at org.jsweet.test.transpiler.AbstractTest.eval(AbstractTest.java:182)
        at org.jsweet.test.transpiler.AbstractTest.eval(AbstractTest.java:177)
        at org.jsweet.test.transpiler.SyntaxTests.testLiterals(SyntaxTests.java:186)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
        at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:283)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:173)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:128)
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
Tests run: 16, Failures: 1, Errors: 0, Skipped: 1, Time elapsed: 72.768 sec <<< FAILURE! - in org.jsweet.test.transpiler.SyntaxTests
testLiterals(org.jsweet.test.transpiler.SyntaxTests)  Time elapsed: 0.108 sec  <<< FAILURE!
java.lang.AssertionError: exception occured while running test SyntaxTests.testLiterals with module kind none
        at org.junit.Assert.fail(Assert.java:88)
        at org.jsweet.test.transpiler.AbstractTest.eval(AbstractTest.java:202)
        at org.jsweet.test.transpiler.AbstractTest.eval(AbstractTest.java:182)
        at org.jsweet.test.transpiler.AbstractTest.eval(AbstractTest.java:177)
        at org.jsweet.test.transpiler.SyntaxTests.testLiterals(SyntaxTests.java:186)
```

**Comment by BurtHarris** [comment]
@renaudpawlak, I traced through the test using Eclipse and it succeeded.   I hate that kind of bug.

I don't know much about Java or Maven, but could Maven running the tests somehow be involved?   

Is there an alternate way to run them?

**Comment by renaudpawlak** [comment]
Yes, @lgrignon told me that it was working under Eclipse. To me, it looks like a Maven command-line launcher problem under Windows (if works fine with Linux and OsX). I did not investigate enough to claim it is a Maven/Windows bug, but I would not be surprised if it was.

**Comment by lgrignon** [comment]
Fixed for 1.2.1 through df2286be5d7ca29dbb968b252ae06317137ea443 and deployed in Maven
